File: Easy/17.ArrayAddition/solve.rb
def ArrayAdditionI(arr)

  # code goes here
  arr.sort!
  m = arr[arr.length-1]
  sub_array = arr[0, arr.length-1]
	
(2..sub_array.length).each do |i|
	combos =  sub_array.combination(i).to_a
	
	combos.each do |combo|
		return true if combo.reduce(0,:+) == m
	end
end
  false
end

# keep this function call here    
puts ArrayAdditionI(STDIN.gets)

File: Medium/29.DistinctList/solve.py
def DistinctList(arr):
	dictionary = { }
	values = list(set(arr))
	for v in values:
		dictionary[v] = 0
	
	for each in arr:
		dictionary[each] += 1
	
	result = 0
	
	for value in dictionary.values():
		if value == 2:
			result += 1
		elif value > 2:
			result += (value-1)
			
	return result
    
# keep this function call here  
print DistinctList(raw_input())